src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,98 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.List;
+
+import duke.dukeexception.DukeException;
+import duke.task.Task;
+import duke.tasklist.TaskList;
+
+
+/**
+ * Represents the storage of taskList of this Duke robot.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Class construtor. Construct a storage according to the file path given.
+     * If there is already a text file in the file path, then create corresponding object.
+     * If there is not, create the directory and the text file.
+     * Throws exception when file operation fails.
+     * @param filepath A string represents the file path of the taskList.
+     * @throws IOException Throws exception when error occurs.
+     */
+    public Storage(String filepath) throws IOException {
+        String s = "./" + filepath.split("duke")[0];
+        File f = new File(s);
+        this.file = new File(s + "duke.txt");
+        if (!this.file.exists()) {
+            f.mkdir();
+            this.file.createNewFile();
+        }
+    }
+    /**
+     * Update the taskList content in the file.
+     * @param t the corresponding list of task that will be written in the file.
+     * @throws DukeException Throws exception when file operation fails.
+     */
+    public void updateFile(List<Task> t) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(this.file);
+            for (int i = 0; i < t.size(); i++) {
+                if (t.get(i) != null) {
+                    if (t.get(i).getStatusIcon().equals("X")) {
```

Comment:
Following code quality guidelines, they don't really like many levels of indentation. Maybe can consider splitting up the if statements so won't have to nest this if statement. Something like:
```
Task task = t.get(i);
if (task == null) {
    continue;
}
if (task.getStatusIcon().equals("X")) {
  ...
} else {
...
}
```